scripts/xpost.coffee

```diff
@@ -12,13 +12,15 @@
 #   No notes needed
 #
 # Author:
-#   @afeld with ideas from @wslack
+#   @afeld and @wslack
 
 module.exports = (robot) ->
   console.log("XPOST script loaded.")
   robot.hear /\bxpost #([\w\-]+)/i, (msg) ->
-    channel = msg.match[1]
-    console.log("cross-posting to ##{channel}")
-
+    target = msg.match[1]
+    poster = msg.message.user.name
     text = msg.message.text
-    robot.messageRoom(channel, text)
+#    console.log("cross-posting to #{target}")
+#    console.log(msg)
+    robot.messageRoom(target, "XPOST from " + poster + " in " + msg.message.room + " -- " + text)
```

Comment:
SGTM +1 shipit :shipit:

On Thu, Oct 15, 2015 at 4:56 PM, Aidan Feldman <notifications@github.com>
wrote:

> In scripts/xpost.coffee
> <https://github.com/18F/18f-bot/pull/26#discussion_r42179356>:
>
> >      text = msg.message.text
> > -    robot.messageRoom(channel, text)
> > +#    console.log("cross-posting to #{target}")
> > +#    console.log(msg)
> > +    robot.messageRoom(target, "XPOST from " + poster + " in " + msg.message.room + " -- " + text)
>
> I think this will basically require live testing. I say we merge it, and
> make an issue, and come back to it later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/18f-bot/pull/26/files#r42179356>.
>